File: projeto 1 - Marcos Barros/cliente.py
# Definição da classe Cliente usando princípios de POO
class Cliente:
    # Construtor da classe que inicializa nome, idade e saldo do cliente
    def __init__(self, nome: str, idade: int, saldo: float):
        # Atributos privados (encapsulados) com dois underlines (__)
        self.__nome = nome
        self.__idade = idade
        self.__saldo = saldo

    # Método para exibir as informações do cliente
    def mostrar_informacoes(self):
        print(f"Cliente: {self.__nome}, Idade: {self.__idade}, Saldo: {self.__saldo:.2f}")

    # Método para atualizar o saldo do cliente (adiciona ou subtrai valor)
    def atualizar_saldo(self, valor: float):
        self.__saldo += valor

    # Métodos getters para acessar os atributos privados (encapsulamento)
    def get_nome(self):
        return self.__nome

    def get_idade(self):
        return self.__idade

    def get_saldo(self):
        return self.__saldo

    # Método setter para atualizar o saldo com validação (exemplo de abstração)
    def set_saldo(self, novo_saldo: float):
        if novo_saldo >= 0:
            self.__saldo = novo_saldo
        else:
            print("Saldo inválido. O saldo não pode ser negativo.")

File: Projeto_1/Projeto_1-LauroLuiz/src/Cliente.ts
// Classe que representa um cliente da loja
export class Cliente{
    // Atributos privados
    private nome: string;
    private idade: number;
    private saldo: number;

    // Construtor para inicializar os atributos
    constructor(nome:string, idade:number, saldo:number){
        this.nome = nome;
        this.idade = idade;
        this.saldo = saldo;
    }

// Método para exibir as informações do cliente
public mostrar_informacoes(){
    console.log(`Cliente: ${this.nome}, Idade: ${this.idade}, Saldo: ${this.saldo}`);
}

    // Método para atualizar o saldo do cliente
    public atualizar_saldo(valor:number){
        this.saldo += valor
    }
}

